### ISSUE TITLE
No event in Debug View with google_analytics_automatic_screen_reporting_enabled = false

### ISSUE THREAD

**Issue by lukas1** [firebase/firebase-android-sdk#3189]
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version:
```
Android Studio Arctic Fox | 2020.3.1 Patch 3
Build #AI-203.7717.56.2031.7784292, built on October 1, 2021
Runtime version: 11.0.10+0-b96-7281165 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 11.6
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Registry: external.system.auto.import.disabled=true
Non-Bundled Plugins: com.phraseapp.androidstudio, Dart, org.jetbrains.kotlin, io.flutter, com.google.mad-scorecard
```
  * Firebase Component: com.google.firebase:firebase-analytics-ktx:20.0.0
  * Component version: 20.0.0

#### Steps to reproduce:

In `AndroidManifest` add this:

```
        <meta-data
            android:name="google_analytics_automatic_screen_reporting_enabled"
            android:value="false"
            />
```

Log somewhere a custom event, using typical code `analytics.logEvent(name, params)`

Enable debugging via command line `adb shell setprop debug.firebase.analytics.app package_name`

Launch the app, observe Debug View.

**Expected behaviour:**
Events should be visible in Debug View. 

**Actual behaviour:**
No events are reported to Debug View. Changing `false` to `true` in manifest will show custom events; If you disable this again in the code, subsequent builds will still log events, until you uninstall the app (maybe there are other ways as well).

This behaviour can also be reproduced on iOS platform with iOS Firebase SDK.

**Comment by lukas1** [comment]
Possibly this issue could be related with this?
https://github.com/firebase/firebase-android-sdk/issues/2225

**Comment by argzdev** [comment]
Hi @lukas1, thanks for reporting. I'll try to investigate this and see what I can find.

**Comment by argzdev** [comment]
Hi @lukas1, I wasn't able to repro the issue. `google_analytics_automatic_screen_reporting_enabled` seems to work fine with either values `true` or `false`. Perhaps you selected a different device on a debug device, I initially had this mistake as well when I was trying to repro your issue.

<img width="205" alt="Screen Shot 2021-11-30 at 9 16 09 PM" src="https://user-images.githubusercontent.com/93124282/144054259-0ea0a5f7-49de-4aa0-89ef-214118ba4129.png">

However, if you're still experiencing this issue, please provide me a [MCVE](https://stackoverflow.com/help/minimal-reproducible-example) so I can investigate this further.

**Comment by lukas1** [comment]
I will try to do so in the upcoming days;

In the meantime, if you had some events in debug view PRIOR to disabling automatic screen reporting, if Firebase considers it the same session, events seems to be sent also on my device, as explained in the description.

Did you try to uninstall the app with debug view active (so that you can see `app_removed` event) and then install again the app with disabled automatic screen reporting?

**Comment by lukas1** [comment]
one more thing. Our team has noticed the exact same behaviour also on iOS, so I'm not sure, if it's issue directly in the SDK. It's unlikely though, that it would be a development mistake in the Android project, if the iOS has the exact same issue.

**Comment by argzdev** [comment]
Yes I did, it looks like after the app encounters the `app_removed` event, the proceeding events are no longer visible on that session, but a new session is added with the event.

Perhaps I'm missing something, here's the steps I tried:
1. Set `google_analytics_automatic_screen_reporting_enabled` to true, log an event -> event is logged on session A
2. Remove app -> `app_removed` is logged on session A
3. Set `google_analytics_automatic_screen_reporting_enabled` to false, install app, log an event -> a new session is added, and the event is logged there

For the iOS issue, may I ask for the issue #? So I can ask our team to investigate this as well.

**Comment by lukas1** [comment]
iOS issue was not yet reported, I am not member of the iOS team. I hope the team will open an issue and I have asked them to link to this issue when they do.

**Comment by lukas1** [comment]
I have some more information; I did not mention this, but I actually used Android Emulator and not a regular device for testing. 

Yesterday I was able to see some events even with the setup that I described. Then this morning, I didn't see any events, when I had automatic reporting disabled. I did not turn it on and instead tried things like clean build, turning the adb shell property on and off and still nothing. 

I used the power button on the emulator to 'lock' the screen. I repeated this two or three times, not really intentionally and lo and behold and now I see events in the debug view.

So I guess there isn't any issue with the automatic screen reporting after all. I'm not sure I'd be able to provide any actionable MVCE where these issues would be visible, it could be related to the fact, that I have not turned off the debugging adb shell property for too long for example, or it could be related to many other things, that are unlikely to show up in MVCE anyway.

It could also very well be an issue of the emulator for what I know.

Thank you for looking into it, sorry to have wasted your time.
